### ISSUE TITLE
helm chart externalIstiod filed is No effect. istio1.23.3

### ISSUE THREAD

**Issue by J4sonZ** [istio/istio#53761]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

helm chart externalIstiod filed is No effect.
set the value，and it will install istiod local
![image](https://github.com/user-attachments/assets/b348ed41-b713-42af-994e-c2d047f35e31)
![image](https://github.com/user-attachments/assets/1107af26-6c97-4e82-8553-5a7363695a40)

### Version

```Text
1.23.3
```

**Comment by J4sonZ** [comment]
helm upgrade --install istiod istio-official/istiod --version 1.23.3 --namespace istio-system  \
--set global.remotePilotAddress=xx.xx.xx.xx,global.istiod.enabled=false,global.externalIstiod=true,istiodRemote.injectionPath="/inject/cluster/worker-4090/net/worker-4090-network"

**Comment by J4sonZ** [comment]
This is notes in helm chart
------------------------------------
    # Configure a remote cluster data plane controlled by an external istiod.
    # When set to true, istiod is not deployed locally and only a subset of the other
    # discovery charts are enabled.
    externalIstiod: false

**Comment by Mariaortizmarquez** [comment]
Glhh

Obtener Outlook para Android<https://aka.ms/AAb9ysg>

________________________________
De: J4sonZ ***@***.***>
Enviado: sábado, noviembre 2, 2024 2:04:26 a. m.
Para: istio/istio ***@***.***>
Cc: Subscribed ***@***.***>
Asunto: Re: [istio/istio] helm chart externalIstiod filed is No effect. istio1.23.3 (Issue #53761)


helm upgrade --install istiod istio-official/istiod --version 1.23.3 --namespace istio-system
--set global.remotePilotAddress=xx.xx.xx.xx,global.istiod.enabled=false,global.externalIstiod=true,istiodRemote.injectionPath="/inject/cluster/worker-4090/net/worker-4090-network"

**Comment by bleggett** [comment]
Yes - `externalIstiod` has been effectively a broken and misdocumented value for some time now.

Remote istiod installations via Helm have been [massively reworked in 1.24](https://github.com/istio/istio/pull/53294), [fixing this specific issue](https://github.com/istio/istio/pull/53542) and actually documenting the installation of a remote `istiod` via Helm as a supported for the first time in https://github.com/istio/istio.io/pull/15459, which is also part of 1.24.

**Comment by Winbobob** [comment]
For Istio <1.24, you need to use `istiod-remote` chart to install istiod remote profile.
